tests/unit-tests/dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -1728,6 +2081,28 @@ TEST(DDS_DCPS_XTypes_DynamicData, Appendable_ReadValueFromArray)
   verify_array_struct(data);
 }
 
+TEST(DDS_DCPS_XTypes_DynamicData, ReadValueFromArrayXCDR1)
+{
+  const XTypes::TypeIdentifier& ti = DCPS::getCompleteTypeIdentifier<DCPS::AppendableArrayStruct_xtag>();
+  const XTypes::TypeMap& type_map = DCPS::getCompleteTypeMap<DCPS::AppendableArrayStruct_xtag>();
+  const XTypes::TypeMap::const_iterator it = type_map.find(ti);
+  EXPECT_TRUE(it != type_map.end());
+
+  XTypes::TypeLookupService tls;
+  tls.add(type_map.begin(), type_map.end());
+  XTypes::DynamicType_rch dt = tls.complete_to_dynamic(it->second.complete, DCPS::GUID_t());
+
+  unsigned char array_struct[] = {
+    0x00,0x00,0x00,0x12, 0x00,0x00,0x00,0x34, // +8=8 int_32a
+    0x00,0x00,0x00,0xff, 0x00,0x00,0x00,0xff,  // +8=16 uint_32a
+    0x01, 0x02 // +2=18 int_8a
+  };
+  ACE_Message_Block msg(256);
+  msg.copy((const char*)array_struct, sizeof(array_struct));
+  XTypes::DynamicData data(&msg, xcdr1, dt);
+  verify_array_struct(data);
+}
+
 TEST(DDS_DCPS_XTypes_DynamicData, Appendable_SkipNestedMembers)
```

Comment:
We need a similar test for XCDR1 appendable. I think the code hasn't handled nested members for XCDR1 yet. Specifically, `skip_all` currently relies on Dheader, in the case of XCDR2 appendable, to skip the whole type (https://github.com/objectcomputing/OpenDDS/blob/master/dds/DCPS/XTypes/DynamicData.cpp#L2242-L2251). For XCDR1 final/appendable, we'd need to skip all members one by one.